test/Microsoft.Extensions.DataIngestion.Tests/MakdownChunker/MarkdownChunker.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.DataIngestion.Tests
+{
+    public sealed class MarkdownChunker : DocumentChunker
+    {
+        private readonly int _headerLevelToSplitOn;
+        private readonly bool _stripHeaders;
+
+
+        public MarkdownChunker(MarkdownHeaderLevel HeaderLevelToSplitOn = MarkdownHeaderLevel.Header3, bool StripHeaders = true)
+        {
+            _headerLevelToSplitOn = (int)HeaderLevelToSplitOn;
+            _stripHeaders = StripHeaders;
+        }
+
+        public override ValueTask<List<Chunk>> ProcessAsync(Document document, CancellationToken cancellationToken = default)
+        {
+            if (document is null) throw new ArgumentNullException(nameof(document));
+
+            string markdown = document.Markdown.ReplaceLineEndings();
+            string[] lines = markdown.Split(Environment.NewLine);
+
+            var lineStack = new Stack<string>(lines.Reverse());
+            return new ValueTask<List<Chunk>>(ParseLevel(lineStack, 1));
+        }
+
+        private List<Chunk> ParseLevel(Stack<string> lines, int markdownHeaderLevel, string context = null, string lastHeader = null)
+        {
+            List<Chunk> chunks = new List<Chunk>();
+
+            StringBuilder sb = new StringBuilder();
+
+            while (lines.Any())
+            {
+                string line = lines.Pop();
+
+                int leadingHashes = CountLeadingHashes(line.Trim());
```

Comment:
Important: Why not consume this information that has already been provided in the strongly typed model? 

https://github.com/adamsitnik/dataingestion/blob/6e6cedce717c7139a620b5bf67ef2dfee2464fa5/src/Microsoft.Extensions.DataIngestion.Abstractions/Document.cs#L62-L64